### ISSUE TITLE
Bug: cannot change publish date of an episode

### ISSUE THREAD

**Issue by StoryHack** [PodcastGenerator/PodcastGenerator#377]
**System Information**
Podcast Generator Version: 3.1
Webserver: Apache
PHP Version: 7.1

Related to bug #375, when you are editing an episode, if you change the publish date and then try to save, you get errors.

**Comment by coldacid** [comment]
I found that copying the contents of my Podcast Generator site on my host to modify my hosting arrangements ended up throwing all my episodes out of order because of PG's use of the media file's modified-time to determine episode order. Not the same bug as this, but very much related.

I plan to add season and episode support to PG, and I'll make sure to use that data to sort and only rely on mtime in case of duplicates. (This makes me wish we had LINQ in PHP though.)

**Comment by StoryHack** [comment]
The publish date really needs to be a separate field in the xml so it can be used to order things correctly.

**Comment by coldacid** [comment]
I agree, but having persistent episode numbers that get emitted into the RSS feed is a higher priority for my show and for the stream that plays it along with others so I'm tackling that first.

**Comment by StoryHack** [comment]
Yep, both are good chanegs that need to happen. Relying on file names and mtime instead of data fields leads to all kinds of issues.

**Comment by flattery103** [comment]
The podcast date isn't based on the file name. It is based on the modification date of the file. The reason you get this error is when you manually upload the files they are with the permissions of that user that uploaded them.

Webservers run on with their own user for security purposes.

All you need to do is make the .mp3 files in the media folder writable by the webserver either by changing the owner to the same owner the webserver uses or change the permissions of the files to globally writable.
In my case I went into the media folder and ran chown www-data:www-data ./*
If I were lazy I could have just did chmod 766 ./*
Either way works.
